akka-cluster-sharding/src/test/scala/akka/cluster/sharding/InactiveEntityPassivationSpec.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding
+
+import akka.actor.{ Actor, ActorRef, Props }
+import akka.cluster.Cluster
+import akka.testkit.{ AkkaSpec, TestProbe }
+import com.typesafe.config.ConfigFactory
+
+object InactiveEntityPassivationSpec {
+  val config = ConfigFactory.parseString("""
+    akka.loglevel = INFO
+    akka.actor.provider = "cluster"
+    akka.remote.netty.tcp.port = 0
+    akka.remote.artery.canonical.port = 0
+    akka.cluster.sharding.passivate-idle-entity-after = 3 s
+    akka.actor.serialize-messages = off
+    """)
+
+  object Passivate
+  object Entity {
+    def props(probe: ActorRef) = Props(new Entity(probe))
+  }
+  class Entity(probe: ActorRef) extends Actor {
+
+    def id = context.self.path.name
+
+    def receive = {
+      case Passivate ⇒
+        probe ! id + " passivating"
+        context.stop(self)
+      case msg ⇒ probe ! msg
+    }
+
+  }
+
+  val extractEntityId: ShardRegion.ExtractEntityId = {
+    case msg: Int ⇒ (msg.toString, msg)
+  }
+
+  val extractShardId: ShardRegion.ExtractShardId = {
+    case msg: Int ⇒ (msg % 10).toString
+  }
+
+}
+
+class InactiveEntityPassivationSpec extends AkkaSpec(InactiveEntityPassivationSpec.config) {
+  import InactiveEntityPassivationSpec._
+
+  "Passivation of inactive entities" must {
+
+    "passivate entities when they haven't seen messages for the configured duration" in {
+      // single node cluster
+      Cluster(system).join(Cluster(system).selfAddress)
+      val probe = TestProbe()
+      val settings = ClusterShardingSettings(system)
+      val region = ClusterSharding(system).start(
+        "myType",
+        InactiveEntityPassivationSpec.Entity.props(probe.ref),
+        settings,
+        extractEntityId,
+        extractShardId,
+        ClusterSharding(system).defaultShardAllocationStrategy(settings),
+        Passivate
+      )
+
+      region ! 1
+      region ! 2
+      Set(
+        probe.expectMsgType[Int],
+        probe.expectMsgType[Int]) should ===(Set(1, 2))
+
+      Thread.sleep(1000)
+      region ! 2
+      probe.expectMsg(2)
+      Thread.sleep(1000)
+      region ! 2
+      probe.expectMsg(2)
```

Comment:
this doesn't test that it is kept alive as long as there are incoming messages

Reply:
but it's configured to stop after 3 s and here it has only waited 2 s

Reply:
Both entities are activated at the "same time" on line 69 and 70, entity 2 keeps getting messages, the two first seconds, while entity 1 does not, we wait 2 seconds to pass the 3 second limit for entity 1 which is stopped, if messages sent to entity 2 was ignored and it would be stopped, it would lead to "2 passivating" being sent to the probe and the probe expects would fail on line 82 or 88,89.

If we wait for 3 seconds 2 would also be passivated and we would see that "2 passivating"

Reply:
I see, sorry